### PR TITLE
[12.0][IMP]Add native notifications via desktop

### PR DESCRIPTION
**Pull request by flotho**
Also fix https://github.com/OCA/web/issues/723

### CODE REVIEW

**Comment by flotho** [comment]
Hi @hbrunn , I would appreciate a review here.

File: web_notify/static/src/js/web_client.js [hunk 1]
```diff
@@ -61,16 +60,26 @@ odoo.define('web_notify.WebClient', function (require) {
...
-                    title: message.title,
-                    message: message.message,
-                    sticky: message.sticky,
-                    className: message.className,
-                }
-            );
+        	var notif_method = 'notify';
+        	var service_class =  'notification' ;
+
+        	if(message.allow_native_notifications){
```
Comment (hbrunn) [inline web_notify/static/src/js/web_client.js:66]: I miss a call to https://developer.mozilla.org/en-US/docs/Web/API/Notification/permission here

File: web_notify/models/res_users.py [hunk 2]
```diff
@@ -80,6 +85,7 @@ def _notify_channel(
             "message": message,
             "title": title,
             "sticky": sticky,
+            "allow_native_notifications": self.allow_native_notifications,
```
Comment (hbrunn) [inline web_notify/models/res_users.py:88]: this makes all the comparisons in tests fail. You need to adapt https://github.com/OCA/web/blob/f56f71155f23ce07bad57b3a7f9d812412ff5e3a/web_notify/tests/test_res_users.py#L24 and below

File: web_notify/static/src/js/services/notification_service.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+odoo.define('web_notify.NotificationService', function (require) {
+"use strict";
+
+ 	require('bus.BusService');
+
+	var NotificationService = require('web.NotificationService');
+
+	NotificationService.include({
+
+		notify: function (params) {
```
Comment (hbrunn) [inline web_notify/static/src/js/services/notification_service.js:10]: please explain this override

File: web_notify/models/res_users.py [hunk 4]
```diff
@@ -32,6 +32,11 @@ def _compute_channel_names(self):
     notify_info_channel_name = fields.Char(compute="_compute_channel_names")
     notify_default_channel_name = fields.Char(compute="_compute_channel_names")
 
+    allow_native_notifications = fields.Boolean(
```
Comment (hbrunn) [inline web_notify/models/res_users.py:35]: I'm not sure if we need this at all - either the user has allowed desktop notifications, then I think we should use them too, or they didn't, and then we don't need this flag

**Review by hbrunn — CHANGES_REQUESTED** [review]
nice feature. Is it true that in v14 we get this for free?

**Comment by flotho** [comment]
> nice feature. Is it true that in v14 we get this for free?

don't really know, I'll test it soon.

**Comment by pedrobaeza** [comment]
Since v13, installing mail_bot, it ask you for that and you have it indeed "for free".

**Comment by flotho** [comment]
maybe I should better check how it has been done in v13 and backport this.

**Pull request closed without merging**